domain/src/Plugin/Condition/Domain.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\domain\Plugin\Condition\Domain.
+ */
+
+namespace Drupal\domain\Plugin\Condition;
+
+use Drupal\Core\Condition\ConditionPluginBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Plugin\Context\ContextDefinition;
+
+/**
+ * Provides a 'Domain' condition.
+ *
+ * @Condition(
+ *   id = "domain",
+ *   label = @Translation("Domain")
+ * )
+ *
+ */
+class Domain extends ConditionPluginBase {
```

Comment:
I think it's the core pattern. I checked the following modules: 
language: Language
node: NodeType
user: UserRole

I also found CurrentThemeCondition in system module, which also looks cleaner to me.

Beside the class name, I'm not quit surre about any context annotations.